### PR TITLE
doc: dt: split into multiple pages and add overarching design goals

### LINKED ISSUE #22557: document guidelines/principles related to DT usage in Zephyr
Document the following:

zephyr's relationship to mainline linux devicetree (bindings should
  be semantically the same whenever possible; dtlib and edtlib should
  not contain Zephyr specific items)

### PR DESCRIPTION
**Pull request by mbolivar**
Zephyr's devicetree documentation has gotten a bit unwieldy, and is missing a single page which defines the overall goals of the effort.

Fix that by splitting it into multiple pages, doing a bit of cleanup on the result, and adding that single page.

Fixes:  #22557

### CODE REVIEW

**Comment by mbolivar** [comment]
> Fix that by splitting it into multiple pages, doing a bit of cleanup on the result, and adding that single page.

Just my attempt based on meeting minutes and other discussions, of course :).

**Review by galak — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]
This is a huge improvement. Thank you.

**Review by carlescufi — APPROVED** [review]
Looks great @mbolivar. Nothing to add here.

File: doc/guides/dts/bindings.rst [hunk 1]
```diff
@@ -0,0 +1,139 @@
+.. _dt-bindings:
+
+Devicetree bindings
+###################
+
+A :ref:`devicetree` on its own is only half the story for describing the
+available hardware devices. Thee tree itself doesn't tell the build system
```
Comment (MaureenHelm) [inline doc/guides/dts/bindings.rst]: s/Thee/The/

File: doc/guides/dts/design.rst [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+In-tree sample applications shall use :ref:`aliases <dt-alias-chosen>` to
+determine which of multiple possible generic devices of a given type will be
+used in the current build. For example, the :ref:`blinky-sample` uses this to
+determine the LED to blink.
+
+Example remaining work
+======================
+
+Zephyr's :ref:`sanitycheck_script` currently uses :file:`board.yaml` files to
```
Comment (nashif) [inline doc/guides/dts/design.rst]: nit: imo this list should be bulleted :)

**Review by MaureenHelm — APPROVED** [review]

File: doc/guides/dts/bindings.rst (hunk 1 above)
Comment (mbolivar) [reply doc/guides/dts/bindings.rst]: Fixed, thank you.

File: doc/guides/dts/design.rst (hunk 2 above)
Comment (mbolivar) [reply doc/guides/dts/design.rst]: I've made all the example lists bulleted. It looked weird to have some but not others bulleted when playing around with this. Also fixed a typo on this page.

**Comment by mbolivar** [comment]
I added a patch to satisfy checkpatch, but don't want to do too much more cleanup work in this PR. I will send a separate PR on top of this one with more cleanups.

**Review by jhedberg — APPROVED** [review]

**Pull request merged**